GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+
+namespace GenericsHomework
+{
+    public class Node<T>
+    {
+        public Node(T init) 
+        {
+            this.data = init;
+            this.Next = this;

```

Comment:
```suggestion
            _Data = init;
            Next = this;
```
You don't need to use `this` here.